### DISCUSSION TITLE
blog/nextjs-fetch-usecase

### DISCUSSION THREAD

**Discussion by giscus** [theodorusclarence/theodorusclarence.com#170] [General]
# blog/nextjs-fetch-usecase

4 Metrics and examples of how to choose the right fetch method for the right case.

https://theodorusclarence.com/blog/nextjs-fetch-usecase

**Comment by theodorusclarence** [comment]
Hello, kindly leave a like (or 5 😄), and comment if you got something to ask! 🚀

**Comment by daliskafroyan** [comment]
absolute madlad, this brings crystal clear clarity of when to use which technique. once again, thanks!

**Reply by theodorusclarence** [reply]
I'm glad it helped!

**Comment by RobinLbt** [comment]
Really cool explanation, thanks a lot. 
NextJS should include something like this in their documentation.

For a SaaS app with gated authentification, it seems like CSR is the way. 
The question is, would you still use NextJS or go with create-react-app ?

**Reply by theodorusclarence** [reply]
Personally, I’ll still go with Next.js. I really like the page routing system

**Comment by surajjizz** [comment]
That's a great explanation and made the concepts clear to the point. Also read the other article on the implementation. Both were fantastic!

**Comment by mikiyasgirma** [comment]
What a brief explanation man, just fucking crazy. so fantastic bruh keep up the good work

**Comment by tomascarignano98** [comment]
Thank you Clarence!

May I ask which software you used for the illustrations? They look great.

**Reply by theodorusclarence** [reply]
I use excalidraw

**Reply by tomascarignano98** [reply]
Thanks mate!

**Comment by benloeffel** [comment]
Thank you, for this detailed and helpful article @theodorusclarence! 👍

**Comment by rvelati** [comment]
Thank you, very interesting, I've got this question, in terms of content delivered to the client, which solution is better  ? in other words which solution delivered less js libraries through the network ? ( I need to improve site CWV results )

**Reply by theodorusclarence** [reply]
Hi, thanks for the interesting question.

To be honest, I haven't looked into the js produced within the build and I'm not that familiar with Core Web Vitals.

I suggest looking for the answer in the [Next.js Discord](https://discord.com/invite/nextjs)

**Comment by amirgholikhani** [comment]
Hi Theodorus
Thanks a lot.
This is very helpful

**Comment by kamesdev** [comment]
Great article!
